autorest/codegen/__init__.py

```diff
@@ -121,9 +121,30 @@ def _build_code_model_options(self) -> Dict[str, Any]:
         """Build en options dict from the user input while running autorest.
         """
         azure_arm = self._autorestapi.get_boolean_value("azure-arm", False)
-        credential_scopes = self._autorestapi.get_value("credential-scopes")
-        if not credential_scopes and azure_arm:
-            credential_scopes = "https://management.azure.com/.default"
+        credential = (
+            self._autorestapi.get_boolean_value("add-credentials", False) or
+            self._autorestapi.get_boolean_value("add-credential", False)
+        )
+
+        credential_scopes_temp = self._autorestapi.get_value("credential-scopes")
+        credential_scopes = credential_scopes_temp.split(",") if credential_scopes_temp else None
+        if credential_scopes and not credential:
+            raise ValueError("--credential-scopes must be used with the --add-credential flag")
+
+
+        if not credential_scopes:
+            if azure_arm:
+                credential_scopes = "https://management.azure.com/.default"
+            elif credential:
+                # If add-credential is specified, we still want to add a credential_scopes variable.
+                # Will make it an empty list so we can differentiate between this case and None
+                _LOGGER.warning(  # pylint: disable=logging-not-lazy
```

Comment:
I added this pylint disable because it doesn't like when you add strings in a logging function.

If I were to remove this answer, I would have to do something along the lines of _LOGGER.warning(%s%s%s, then the string fragments).

Reply:
It's the `+` the problem, and it's not needed
```
In [1]: import logging

In [2]: logger = logging.getLogger()

In [3]: logger.warning(
   ...:     "foo"
   ...:     "bar"
   ...: )
foobar
```

Python allows multiline string and do auto-concatenation

Reply:
Oh, I didn't know that, thanks so much!